src/controls/LevelingSensor/QskLevelingSensorSkinlet.cpp

```diff
@@ -0,0 +1,461 @@
+    auto* const bNode = static_cast< QskBoxNode* >( root->firstChild() );
+
+    const auto size = outerRadius( sensor ) * sensor->strutSizeHint( Q::OuterDisk ).width();
+    updateBoxNode( sensor, bNode, { 0, 0, 2 * size, 2 * size }, boxShapeMetrics, boxBorderMetrics,
+        boxBorderColors, boxGradient );
+
+    const auto cX = center( sensor ).x();
+    const auto cY = center( sensor ).y();
+    const auto rZ = 0.0;
+
+    const auto matrix = matrix_deg( 0.0, 0.0, 0.0, cX, cY, 0 ) *
+                        matrix_deg( 0.0, 0.0, rZ, 0, 0, 0 ) *
+                        matrix_deg( 0.0, 0.0, 0.0, -size, -size, 0 );
+
+    root->setMatrix( matrix );
+    return root;
+}
+
+template<>
+QSGNode* QskLevelingSensorSkinlet::updateSubNode< R::Horizon >(
+    const QskLevelingSensor* const sensor, const quint8 nodeRole, QSGNode* const node ) const
+{
+    const auto subControl = Q::Horizon;
+    const State<QskAspect::Subcontrol> state(sensor, subControl);
+    
+    const auto dY = 2 * sensor->angle().y();
+    const auto pY = qBound( 0.0, 0.5 + ( -state.rX / dY ), 1.0 );
+
+    const auto shape = QskBoxShapeMetrics{ state.r1 };
+    const auto metrics = sensor->boxBorderMetricsHint( subControl );
+    const auto colors = sensor->boxBorderColorsHint( subControl );
+
+    auto gradient = sensor->gradientHint( Q::Horizon );
+    gradient.setDirection( QskGradient::Linear );
+    gradient.setLinearDirection( Qt::Vertical );
+    gradient.setStops( { { 0.0, gradient.startColor() }, { pY, gradient.startColor() },
+        { pY, gradient.endColor() }, { 1.0, gradient.endColor() } } );
+
+    auto* const tNode = ensureNode< QSGTransformNode, QskBoxNode >( node );
+    auto* const boxNode = static_cast< QskBoxNode* >( tNode->firstChild() );
+    updateBoxNode( sensor, boxNode, { 0, 0, 2 * state.r1, 2 * state.r1 }, shape, metrics, colors, gradient );
+
+    const auto matrix = matrix_deg( 0, 0, 0, state.cX, state.cY, 0 ) *
+                        matrix_deg( 0, 0, 0, -state.r1, -state.r1, 0 );
+
+    tNode->setMatrix( matrix );
+    return tNode;
+}
+
+template<>
+QSGNode* QskLevelingSensorSkinlet::updateSubNode< R::TickmarksX >(
+    const QskLevelingSensor* const sensor, const quint8 nodeRole, QSGNode* const node ) const
+{
+    const auto subControl = Q::TickmarksX;
+    State<QskAspect::Subcontrol> state(sensor, subControl);
+    const auto color = sensor->color( subControl );
+    const auto scale = sensor->strutSizeHint( subControl );
+    const auto r3 = state.r1 * scale.height();
+
+    auto* const clipping =
+        ensureNode< PolygonClipNode, QSGTransformNode, LinearTickmarksNode >(
+            node );
+    auto* const transform = static_cast< QSGTransformNode* >( clipping->firstChild() );
+    auto* const tickmarks = static_cast< LinearTickmarksNode* >( transform->firstChild() );
+
+    auto size = qvariant_cast< QVector3D >( sensor->effectiveSkinHint( subControl ) ) * r3;
```

Comment:
My fist idea would be to use QskAspect::Variation, where you have Tiny/Small/Large/Huge. However this would block using other variations like Vertical/Horizontal. So maybe it is not a bad idea to introduce a class for a tuple of increasing sizes - in the best case something that works for other situations as well. 

Of course QVector3D works out of the box, but the QVectorXD classes are float ( not qreal ) based and have primarily been made for OpenGL APIs.